tests/Feature/UserRepositoryTest.php

```diff
@@ -821,4 +804,361 @@ public function testGetAllUsers()
+        "Initial-Connection","user_id":"507f1f77bcf86cd799439021","provider":"auth0","isSocial":false}],
+        "app_metadata":{},"user_metadata":{},"picture":"","name":"","nickname":"","multifactor":[""],"last_ip":
+        "","last_login":"","logins_count":0,"blocked":false,"given_name":"","family_name":""}]');
+        $response2 = new Response(200, [], '[{"user_id":"third_user",
+        "email":"three@gmail.com","email_verified":false,"username":"user3","phone_number":
+        "+199999999999999","phone_verified":false,"created_at":"","updated_at":"","identities":[{"connection":
+        "Initial-Connection","user_id":"507f1f77bcf86cd799439022","provider":"auth0","isSocial":false}],
+        "app_metadata":{},"user_metadata":{},"picture":"","name":"","nickname":"","multifactor":[""],"last_ip":
+        "","last_login":"","logins_count":0,"blocked":false,"given_name":"","family_name":""}]');
+        $this->mockedResponses = [$response1, $response2];
+
+        // Set chunk size to 2 in the config, and create new repository using this chunk size.
+        Config::set('mrd-auth0.chunk_size', 2);
+        $this->resetRepo();
+
+        // Call function under test.
+        $users = $this->repo->getByIds(collect(['auth0|507f1f77bcf86cd799439020', 'other_user', 'third_user']));
+
+        // Expect three users returned.
+        self::assertEquals(3, $users->count());
+        // Verify keyed by user id.
+        self::assertContains('auth0|507f1f77bcf86cd799439020', $users->keys());
+        self::assertContains('other_user', $users->keys());
+        self::assertContains('third_user', $users->keys());
+
+        // Verify three users
+        $user1 = $users->get('auth0|507f1f77bcf86cd799439020');
+        self::assertNotNull($user1);
+        self::assertEquals('johndoe', $user1->username);
+
+        $user2 = $users->get('other_user');
+        self::assertNotNull($user2);
+        self::assertEquals('other', $user2->username);
+
+        $user3 = $users->get('third_user');
+        self::assertNotNull($user3);
+        self::assertEquals('user3', $user3->username);
+
+        // Expect 2 api calls.
+        self::assertCount(2, $this->guzzleContainer);
+
+        // Find the request that was sent to Auth0
+        $request1 = $this->guzzleContainer[0]['request'];
+        $request2 = $this->guzzleContainer[1]['request'];
+        $query1 = urldecode($request1->getUri()->getQuery());
+        $query2 = urldecode($request2->getUri()->getQuery());
+
+        // Verify correct endpoint was called.
+        self::assertEquals('/api/v2/users', $request1->getUri()->getPath());
+        self::assertEquals('/api/v2/users', $request2->getUri()->getPath());
+        // Verify correct query sent to Auth0. Order of IDs does not matter.
+        self::assertTrue(
+            str_contains($query1, 'q=user_id:("auth0|507f1f77bcf86cd799439020" OR "other_user")') ||
```

Comment:
Or is used in this test because the order in which the user IDs are placed in the query does not matter, so either the query should be `user_id:(user1 OR user2)` or it should be `user_id:(user2 OR user1)`. 
Your suggestion will fail always since the query should not include both orderings.

Reply:
Ahh I see. I was misinterpreting the 'OR' inside the string as the OR that either field could be in there. Makes sense.